### PR TITLE
#576 - Repository 'releases' tab missing some releases

### PR DESCRIPTION
**Pull request by LetItRock**
General changes:
- RepositoryTagsFragment, TagsAdapter and RepositoryTagsPresenter
- For every tags we also receive information about release if exist
- AbstractCacheDataSource - for caching some data (not only tags)
- TagsCacheDataSource, TagsCloudDataSource, TagsRetrofitWrapper
- Service for receiving tags, commits and releases - per_page for tags is 15
- Changed tags adapter for RecyclerView in RepositoryTagsFragment - now it displaying tags and releases, lazy load count is 5
- RepositoryTagsFragment - added CommitInfoPresenter - because we want to fetch commit info from single tag
- Material dialog for tags and releases - created custom view for tag details dialog
- Dependency Injection and IoC for RepositoryTagsPresenter
- Schedulers for UI thread and IO thread could be now injected
- Token could be also injected using @Inject and @Token annotations
- Dagger 2 dependencies for tests
- Test for RepositoryTagsPresenter
- Dependency for retrofit-mock
- Tests for TagsCloudDataSource
- BeanUtils - class for creating java beans for test purposes
- Dto's
- Beans

### CODE REVIEW

**Comment by alorma** [comment]
Good job! I will check deeply later!

**Comment by pkramme** [comment]
Holy shit. That's huge!

**Comment by MelvynEzi** [comment]
This dude is da man

**Comment by alorma** [comment]
Could you provide an apk to test it?

**Comment by LetItRock** [comment]
[app-debug.zip](https://github.com/gitskarios/Gitskarios/files/488453/app-debug.zip)
Here you go... please unzip it...

File: app/src/main/java/com/alorma/github/injector/component/ApiComponent.java [hunk 1]
```diff
@@ -48,4 +51,7 @@
   void inject(CommitDetailActivity commitDetailActivity);
 
   void inject(UserIssuesListFragment userIssuesListFragment);
+
+  // next components with its own tree
+  RepositoryTagsComponent plus(RepositoryTagsModule module);
```
Comment (alorma) [inline app/src/main/java/com/alorma/github/injector/component/ApiComponent.java:56]: `plus` ? Better a descriptive name

File: app/src/main/java/com/alorma/github/injector/module/tags/RepositoryTagsModule.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import com.alorma.github.sdk.core.ApiClient;
+import com.alorma.github.sdk.core.repositories.releases.tags.TagsCloudDataSource;
+import com.alorma.github.sdk.core.repositories.releases.tags.TagsRetrofitWrapper;
+
+import dagger.Module;
+import dagger.Provides;
+import rx.Scheduler;
+
+@Module
+public class RepositoryTagsModule {
```
Comment (alorma) [inline app/src/main/java/com/alorma/github/injector/module/tags/RepositoryTagsModule.java:19]: 😻

**Comment by alorma** [comment]
Great!!!!!

**Pull request merged**